app/views/events/show.html.erb

```diff
@@ -2,23 +2,34 @@
   <h1>Event Title: <%= @event.title %></h1>
   <div class="row">
   	<div class="col-md-6">
-	  <%= form_for [@event, @registration] do |f| %>
-	    <%= f.hidden_field :event_id %>
-	    <%= f.hidden_field :user_id %>
-	    <%= f.hidden_field :leader, value: '0' %>
-	    <% if current_user.is_leader? %>
-	      <%= f.label 'Register as Leader' %>
-	      <%= f.check_box :leader %>
-	    <% end %>
-	    <%= f.submit %>
-	  <% end %>
+      <% if @current_user %>
```

Comment:
I think the right way to access this is just current_user (function in devise stuff) rather than the @variable that backs the function.